web/react/components/create_post.jsx

```diff
@@ -112,6 +131,63 @@ module.exports = React.createClass({
     handleUserInput: function(messageText) {
         this.resizePostHolder();
         this.setState({messageText: messageText});
+
+        // look to see if the message begins with any carets to indicate that it's a reply
+        var replyMatch = messageText.match(/^\^+/g);
+        if (replyMatch) {
+            // the number of carets indicates how many message threads back we're replying to
```

Comment:
That's a good point. I'm doing a bit of that by not sending the events out if the rootId doesn't change, but it's still reparsing the message each time if it does begin with carets. I need to think about how that'll work when someone else sends a message while you're typing one though since we might want it to switch what you're replying to when that happens.